compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4127,7 +4140,8 @@ public BType getTypeIntersection(IntersectionContext intersectionContext, BType
         }
     }
 
-    private BType getIntersection(IntersectionContext intersectionContext, BType lhsType, SymbolEnv env, BType type) {
+    private BType getIntersection(IntersectionContext intersectionContext, BType lhsType, SymbolEnv env, BType type,
```

Comment:
The intersection computation still seems to have issues, although it does not crash.

```ballerina
type Foo int|Foo[]|object {};
type Bar int|Bar[]|string;

public function main() {
    Foo s = <int[]> [1, 2];
    float|Foo m = <Foo>s;
    if (m is Bar|string) {
        int f2 = m; // no compilation error, panics at runtime
        panic error("1");
    }
    panic error("2");
}
```

Reply:
This is an existing issue right